curriculum/challenges/english/25-front-end-development/review-loops-and-sequences/67f39b91583dec1e1eddbb9e.md

```diff
@@ -7,13 +7,739 @@ dashedName: review-loops-and-sequences
+- Here is an example of using the `key` argument to sort items in a tuple by length:
+
+```py
+programming_languages = ('Rust', 'Java', 'Python', 'C++', 'Rust', 'Python')
+sorted(programming_languages, key=len)
+
+# Result
+# ['C++', 'Rust', 'Java', 'Rust', 'Python', 'Python']
+```
+
+- You can create a new list of values in reverse order, using the `reverse` argument like this:
+
+```py
+programming_languages = ('Rust', 'Java', 'Python', 'C++', 'Rust', 'Python')
+
+print(sorted(programming_languages, reverse=True))
+
+# Result
+# ['Rust', 'Rust', 'Python', 'Python', 'Java', 'C++']
+```
+
+## Loops in Python
+
+- **Definition**: Loops are used to repeat a block of code for a set number of times.
+
+- **`for` loop**: Used to iterate over a sequence (like a list, tuple or string) and execute a block of code for each item in that sequence.
+- Here is an example of using a `for` loop to iterate through a list and print each language to the console:
+
+```py
+programming_languages = ['Rust', 'Java', 'Python', 'C++']
+
+for language in programming_languages:
+    print(language)
+
+"""
+Result 
+
+'Rust'
+'Java'
+'Python'
+'C++'
+"""'''
+```
+
+- Here is an example of using a `for` loop to loop through the string `'code'` and print out each character:
+
+```py
+for char in 'code':
+    print(char)
+
+"""
+Result 
+
+'c'
+'o'
+'d'
+'e'
+"""
+```
+
+- `for` loops can be nested. 
+- Here is an example of using a nested `for` loop:
+
+```py
+categories = ['Fruit', 'Vegetable']
+foods = ['Apple', 'Carrot', 'Banana']
+
+for category in categories:
+    for food in foods:
+        print(category, food)
+
+"""
+Result
+
+Fruit Apple
+Fruit Carrot
+Fruit Banana
+Vegetable Apple
+Vegetable Carrot
+Vegetable Banana
+"""
+
+```
+
+- **`while` loop**: Repeats a block of code until the condition is `False`. 
+
+- Here is an example of using a `while` loop for a guessing game:
+
+```py
+secret_number = 3
+guess = 0
+
+while guess != secret_number:
+    guess = int(input('Guess the number (1-5): '))
+    if guess != secret_number:
+        print('Wrong! Try again.')
+
+print('You got it!')
+
+"""
+Result
+
+Guess the number (1-5): 2
+Wrong! Try again.
+Guess the number (1-5): 1
+Wrong! Try again.
+Guess the number (1-5): 3
+You got it!
+"""
+
+```
+
+- **`break` and `continue` statements**: Used in loops to modify the execution of a loop.
+
+- The `break` statement is used to exit the loop immediately when a certain condition is met.
+
+Here is an example of using the `break` statement for a list of `developer_names`:
+
+```py
+developer_names = ['Jess', 'Naomi', 'Tom']
+
+for developer in developer_names:
+    if developer == 'Naomi':
+        break
+    print(developer)
+```
+
+- The `continue` statement is used to skip that current iteration and move onto the next iteration of the loop. 
+
+Here is an example to use the `continue` statement instead of a `break` statement:
```

Comment:
```suggestion
- The `continue` statement is used to skip that current iteration and move onto the next iteration of the loop.  Here is an example to use the `continue` statement instead of a `break` statement:
```